src/Core/Services/MultipleMutationInputValidator.cs

```diff
@@ -0,0 +1,575 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Net;
+using Azure.DataApiBuilder.Config.DatabasePrimitives;
+using Azure.DataApiBuilder.Config.ObjectModel;
+using Azure.DataApiBuilder.Core.Resolvers;
+using Azure.DataApiBuilder.Core.Services.MetadataProviders;
+using Azure.DataApiBuilder.Service.Exceptions;
+using Azure.DataApiBuilder.Service.GraphQLBuilder;
+using Azure.DataApiBuilder.Service.Services;
+using HotChocolate.Language;
+using HotChocolate.Resolvers;
+
+namespace Azure.DataApiBuilder.Core.Services
+{
+    public class MultipleMutationInputValidator
+    {
+        /// <summary>
+        /// Recursive method to validate a GraphQL value node which can represent the input for item/items
+        /// argument for the mutation node, or can represent an object value (*:1 relationship)
+        /// or a list value (*:N relationship) node for related target entities.
+        /// </summary>
+        /// <param name="schema">Schema for the input field</param>
+        /// <param name="entityName">Current entity's name.</param>
+        /// <param name="context">Middleware Context.</param>
+        /// <param name="parameters">Value for the input field.</param>
+        /// <param name="runtimeConfig">Runtime config</param>
+        /// <param name="columnsDerivedFromParentEntity">Set of columns in this entity whose values should be
+        /// derived from insertion in the source entity (i.e. source entity would have been the referenced entity).</param>
+        /// <param name="columnsToBeDerivedFromEntity">Set of columns in this entity whose values are to be
+        /// derived from insertion in the this entity and returned to the source entity so as to provide values for
+        /// the corresponding referencing fields (i.e. source entity would have been the referencing entity).</param>
+        /// <param name="nestingLevel">Current depth of nesting in the multiple-create request.</param>
+        /// <param name="parentEntityName">Parent entity's name.</param>
+        /// <param name="sqlMetadataProviderFactory">Metadata provider factory.</param>
+        /// <example>       1. mutation {
```

Comment:
These examples look like valid requests. If the intention was to outline few examples of invalid requests which the method will catch successfully, adding a one-liner as to why the request is invalid will help.

Reply:
This example is to show the catgeorisation as to which input will be processed by which IF/ELSEIF block. Similar to what was done in authz processing. There are a lot of invalid scenarios which can occur and to provide examples for each of them here will be an overkill I think because I will then have to explain as well, why the request is invalid which I think is best handled when each of the invalid scenarios are actually validated.